packages/client/src/components/ViewLogEntries/ViewLogEntriesTable.tsx

```diff
@@ -1,87 +1,112 @@
-import { LogEntryResponse } from '@mapistry/take-home-challenge-shared';
+import {
+	DateLike,
+	EditLogEntryRequest,
+	LogEntryResponse,
+} from '@mapistry/take-home-challenge-shared';
 import { useCallback } from 'react';
 import styled from 'styled-components';
 import { useLogEntries } from '../../hooks/useLogEntries';
 import { deleteLogEntry } from '../../shared/apiClient/logsApi';
 
 interface ViewLogEntriesTableProps {
-  logId: string;
+	logId: string;
+	setIsEditEntryOpen: (value: boolean) => void;
+	setInitialEditValues: (value: EditLogEntryRequest) => void;
 }
 
 const StyledTable = styled.table`
-  width: 100%;
-  tr:nth-child(even) {
-    background-color: #f2f2f2;
-  }
-  th,
-  td {
-    padding: 1rem;
-    text-align: center;
-  }
-  th {
-    font-weight: 700;
-    border-bottom: 1px solid;
-  }
+	width: 100%;
+	tr:nth-child(even) {
+		background-color: #f2f2f2;
+	}
+	th,
+	td {
+		padding: 1rem;
+		text-align: center;
+	}
+	th {
+		font-weight: 700;
+		border-bottom: 1px solid;
+	}
 `;
 
-export function ViewLogEntriesTable({ logId }: ViewLogEntriesTableProps) {
-  const { logEntries, refreshLogEntries } = useLogEntries({ logId });
-  const handleDelete = useCallback(
-    async (logEntry) => {
-      // eslint-disable-next-line no-restricted-globals, no-alert
-      if (confirm('Are you sure?')) {
-        await deleteLogEntry(logEntry);
-        refreshLogEntries();
-      }
-    },
-    [refreshLogEntries],
-  );
+export function ViewLogEntriesTable({
+	logId,
+	setIsEditEntryOpen,
+	setInitialEditValues,
+}: ViewLogEntriesTableProps) {
+	const { logEntries, refreshLogEntries } = useLogEntries({ logId });
+	const handleDelete = useCallback(
+		async (logEntry) => {
+			// eslint-disable-next-line no-restricted-globals, no-alert
+			if (confirm('Are you sure?')) {
+				await deleteLogEntry(logEntry);
+				refreshLogEntries();
+			}
+		},
+		[refreshLogEntries]
+	);
+	const handleEdit = useCallback(async (logEntry) => {
+		setInitialEditValues({
+			id: logEntry.id,
+			logId: logEntry.logId,
+			logDate: new Date(logEntry.logDate),
+			logValue: logEntry.logValue,
+		});
+		setIsEditEntryOpen(true);
+	}, []);
```

Comment:
I'm wondering if this might need setInitialEditValues or setIsEditEntryOpen in here but maybe not since logEntry is the only dynamic value?